### PR TITLE
phase 2 changes

### PR DESCRIPTION
**Pull request by elaz-loops**
support landscape and toggle bg black/white on tap for light mode

### CODE REVIEW

File: Sources/MediaViewer/Extensions/WorkaroundNavigationController.swift [hunk 1]
```diff
@@ -30,6 +30,12 @@ open class WorkaroundNavigationController: UINavigationController {
         }
         setToolbarHidden(!isToolbarHidden, animated: true)
         setNavigationBarHidden(!isNavigationBarHidden, animated: true)
+        let hidden = isNavigationBarHidden
```
Comment (MaayanLoop) [inline Sources/MediaViewer/Extensions/WorkaroundNavigationController.swift]: nit: don't we need this thing only inside the "if"?

File: Sources/MediaViewer/PreviewItemView/Builtins/ImagePreviewItemViewController.swift [hunk 2]
```diff
@@ -4,14 +4,19 @@ import UIKit
 public final class ImagePreviewItemViewController: UIViewController, UIScrollViewDelegate {
 
     let scrollView = UIScrollView()
-    let imageView = UIImageView()
+    let imageView: UIImageView
     let doubleTapGesture = UITapGestureRecognizer()
-    let image: UIImage
+    private var image: UIImage?
```
Comment (MaayanLoop) [inline Sources/MediaViewer/PreviewItemView/Builtins/ImagePreviewItemViewController.swift]: Why are we keeping this var?

File: Sources/MediaViewer/Transition/Animator/PresentAnimator.swift [hunk 3]
```diff
@@ -22,9 +22,11 @@ final class PresentAnimator: Animator {
               let transitionImage = transitionImageView.image,
               let fromView = fromController?.view,
               let topView = previewController.topView else {
+            
+            (container.viewWithTag(PresentationConsts.transitionViewTag) as? UIImageView)?.removeFromSuperview()
```
Comment (MaayanLoop) [inline Sources/MediaViewer/Transition/Animator/PresentAnimator.swift]: nit: does it matter if it's a UIImageView here? Do we need the casting?

File: Sources/MediaViewer/UI/PreviewController.swift [hunk 4]
```diff
@@ -66,6 +66,29 @@ open class PreviewController: UIViewController {
...
+    open override var shouldAutorotate: Bool {
+        return true
+    }
+    
+    open override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        setOrientation(to: .portrait)
+    }
+    
+    func setOrientation(to orientation: UIInterfaceOrientation) {
```
Comment (MaayanLoop) [inline Sources/MediaViewer/UI/PreviewController.swift]: Does this needs to be public? Should it be in some util class like Device.swift?

File: Sources/MediaViewer/UI/PreviewController.swift [hunk 5]
```diff
@@ -241,3 +264,26 @@ fileprivate final class Toolbar: UIToolbar {
...
+        case .portraitUpsideDown:
+            return .portraitUpsideDown
+        default:
+            return .all
+        }
+    }
+}
+
+internal extension UIViewController {
+    var isDarkMode: Bool {
```
Comment (MaayanLoop) [inline Sources/MediaViewer/UI/PreviewController.swift:281]: In the main app we are separating appDarkMode from deviceDarkMode. Did you check that dark mode is working if I change that setting both from the app settings / device settings?

**Review by MaayanLoop — APPROVED** [review]

File: Sources/MediaViewer/PreviewItemView/Builtins/ImagePreviewItemViewController.swift (hunk 2 above)
Comment (elaz-loops) [reply Sources/MediaViewer/PreviewItemView/Builtins/ImagePreviewItemViewController.swift]: removed

File: Sources/MediaViewer/UI/PreviewController.swift (hunk 4 above)
Comment (elaz-loops) [reply Sources/MediaViewer/UI/PreviewController.swift]: made private, but I see no reason to make it a util in this package. it's specific to this controller which is the top container of preview and not used elsewhere, nor will it be.

File: Sources/MediaViewer/UI/PreviewController.swift (hunk 5 above)
Comment (elaz-loops) [reply Sources/MediaViewer/UI/PreviewController.swift:281]: yes, it does (the check is against the traitCollection, which gets set either from window.overrideUserInterfaceStyle, or from the device's dark mode settings. either way, it's up to date.

**Pull request merged**